rundeckapp/src/test/groovy/rundeck/controllers/ScheduledExecutionControllerSpec.groovy

```diff
@@ -3909,4 +3925,188 @@ class ScheduledExecutionControllerSpec extends RundeckHibernateSpec implements C
         'https://web.server/geto'                                   |_
         'http://web.server/geto'                                    |_
     }
+
+    def "test remote url basic auth"() {
+        given:
+        controller.apiService = Mock(ApiService)
+        controller.frameworkService = Mock(FrameworkService)
+
+        def url = 'http://web.server/geto'
+        def client = Mock(HttpClient)
+        def httpClientCreator = Mock(HttpClientCreator){
+            createClient()>>client
+        }
+
+        def jobChangeLogger = Mock(Logger)
+        controller.logger = jobChangeLogger
+
+        JobOptionConfigRemoteUrl configRemoteUrl = new JobOptionConfigRemoteUrl()
+        configRemoteUrl.authenticationType = RemoteUrlAuthenticationType.BASIC
+        configRemoteUrl.username = "test"
+        configRemoteUrl.password ="test"
+
+        when:
+        def result = controller.getRemoteJSON(httpClientCreator, url, configRemoteUrl, 100, 100, 5,false)
+
+        then:
+        1 * client.setUri(_)
+        1 * client.setBasicAuthCredentials("test","test")
+        1 * client.execute({ RequestProcessor processor->
+            HttpResponse rsp = Mock(HttpResponse){
+                getStatusLine()>>Mock(StatusLine){
+                    getStatusCode()>>200
+                }
+                getEntity()>>Mock(HttpEntity){
+                    getContent()>>new ByteArrayInputStream('{}'.getBytes());
+                }
+                getFirstHeader("Content-Type")>>Mock(Header){
+                    getValue()>>"application/json"
+                }
+            }
+            processor.accept(rsp)
+        })
```

Comment:
so Spock will evaluate this by expecting the closure inside the method call argument to work like a constraint and evaluates the [statements like assertions](https://spockframework.org/spock/docs/2.1/interaction_based_testing.html#_code_constraint) (in this case it still succeeds), but for side-effects like calling processor.accept here, you probably want to do it like this here and in the other tests: 

```suggestion
        1 * client.execute(_) >> {
            RequestProcessor processor = it[0]
            HttpResponse rsp = Mock(HttpResponse){
                getStatusLine()>>Mock(StatusLine){
                    getStatusCode()>>200
                }
                getEntity()>>Mock(HttpEntity){
                    getContent()>>new ByteArrayInputStream('{}'.getBytes());
                }
                getFirstHeader("Content-Type")>>Mock(Header){
                    getValue()>>"application/json"
                }
            }
            processor.accept(rsp)
        }
```